osutil/user.go

```diff
@@ -204,20 +211,35 @@ func AddUser(name string, opts *AddUserOptions) error {
 		return fmt.Errorf("cannot add user %q: name contains invalid characters", name)
 	}
 
-	cmdStr := []string{
-		"adduser",
-		"--force-badname",
-		"--gecos", opts.Gecos,
-		"--disabled-password",
+	var userTool string
+	var cmdStr []string
+	if hasAddUserExecutable() {
```

Comment:
Preference comment: 
Consider keeping orig line 207-211 definition of as default and `if !hasAddUserExecutable() {` for the new alternative?
Perhaps `cmdStr[0]` instead of `userTool` below?

Reply:
ah nice, I removed `userTool` than you for spotting this!
I also reverse the logic as proposed

Reply:
> Preference comment: Consider keeping orig line 207-211 definition with values as default and `if !hasAddUserExecutable() {` for the new alternative (without the `else`)? Perhaps `cmdStr[0]` instead of `userTool` below?

My intention with the proposed change was that this `} else {` goes away, and [lines 207 to 211](https://github.com/snapcore/snapd/pull/13236/files#diff-0ed6946becf5975a25682a072650282bf4258e8e07f154cb713a489fddf66a33L207-L211) that declares `cmdStr := ...` with values comes back.